cluster_pack/packaging.py

```diff
@@ -368,19 +369,40 @@ def detect_archive_names(
         pex_file = ""
         env_name = packer.env_name()
 
+    def build_package_path(name: str = env_name,
+                           extension: Optional[str] = packer.extension()) -> str:
+        path = (f"{get_default_fs()}/user/{getpass.getuser()}"
+                f"/envs/{name}")
+        if extension is None:
+            return path
+        return f"{path}.{extension}"
+
     if not package_path:
-        package_path = (f"{get_default_fs()}/user/{getpass.getuser()}"
-                        f"/envs/{env_name}.{packer.extension()}")
+        package_path = build_package_path()
     else:
         if "".join(os.path.splitext(package_path)[1]) != f".{packer.extension()}":
             raise ValueError(f"{package_path} has the wrong extension"
                              f", .{packer.extension()} is expected")
 
+    # we are actually building or reusing a large pex and we have the information from the
+    # allow_large_pex flag
     if (packer.extension() == PEX_PACKER.extension()
             and allow_large_pex
             and not package_path.endswith('.zip')):
         package_path += '.zip'
 
+    # We are running from an unzipped large pex and we have the information because `pex_file` is
+    # not empty, and it is a directory instead of a zipapp
+
+    if (pex_file != ""
+            and os.path.isdir(pex_file)
+            and not package_path.endswith('.zip')):
+
+        pex_files = glob.glob(f"{os.path.dirname(pex_file)}/*.pex.zip")
+        assert len(pex_files) == 1, \
+            f"Expected to find single zipped PEX in same dir as {pex_file}, got {pex_files}"
+        package_path = build_package_path(os.path.basename(pex_files[0]), None)

```

Comment:
So if I understand this should be the path on hdfs of the large pex created previously in the workflow ?

Reply:
No, this is the path on the local filesystem of the zip (the current executable is the __main__.py in the unzipped pex)